pyproject.toml

```diff
@@ -38,7 +38,7 @@ dependencies = [
     "numpy >=1.20",
     "asdf >=2.8.2,!=2.14.0,!=2.14.1,!=2.14.2",
     "pandas >=1.0",
-    "xarray >=0.19,<2022.09.0",
+    "xarray >=2023.3.0",
```

Comment:
Actually we can also think of finding the XArray version which introduced Dataset.merge(join="override"), but I found it easier to just request at least a recent version.